### PR TITLE
Show bindings when assertions fail in ExUnit

### PR DESCRIPTION
**Pull request by whatyouhide**
This PR shows the value of non-root-level variables in asserted expressions when an assertion fails:

screen shot 2017-05-31 at 12 52 01 pm

### CODE REVIEW

File: lib/ex_unit/test/ex_unit/formatter_test.exs [hunk 1]
```diff
@@ -136,6 +136,22 @@ defmodule ExUnit.FormatterTest do
     """
   end
 
+  test "shows bindings" do
```
Comment (josevalim) [inline lib/ex_unit/test/ex_unit/formatter_test.exs]: `test "formats bindings"` ?

File: lib/ex_unit/lib/ex_unit/formatter.ex [hunk 2]
```diff
@@ -222,6 +224,11 @@ defmodule ExUnit.Formatter do
     code_multiline(Macro.to_string(expr), padding_size)
   end
 
+  defp format_bindings(bindings) do
+    padding = "\n" <> @counter_padding <> "  "
+    padding <> Enum.map_join(bindings, padding, fn {var, value} -> "#{var} = #{inspect(value)}" end)
```
Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/formatter.ex]: You may want to look into using inspect_multiline.

File: lib/ex_unit/lib/ex_unit/formatter.ex [hunk 3]
```diff
@@ -222,6 +224,19 @@ defmodule ExUnit.Formatter do
...
   end
 
+  defp format_binding(binding, width) do
+    padding = @counter_padding <> "  "
+    padding_size = byte_size(padding)
+
+    result =
+      Enum.map_join(binding, "\n" <> padding, fn {var, value} ->
+        first_part = "#{var} = "
+        first_part <> inspect_multiline(value, padding_size + String.length(first_part), width)
```
Comment (josevalim) [inline lib/ex_unit/lib/ex_unit/formatter.ex]: I don't think we should do `+ String.length`. For long variable names we will have very little space. I think indenting two spaces past the variable name will be fine.

Comment (whatyouhide) [inline lib/ex_unit/lib/ex_unit/formatter.ex]: Oh, okay sure.

**Comment by ericmj** [comment]
Can we include the macro in code so that "code: assert my_starts_with?..."?

**Pull request merged**